tests/ssl-opt.sh

```diff
@@ -9044,6 +9090,10 @@ run_test    "SSL async private: error in resume then operate correctly" \
 # key1: ECDSA, key2: RSA; use key1 through async, then key2 directly
 requires_config_enabled MBEDTLS_SSL_ASYNC_PRIVATE
 requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
+# Note: the function "detect_required_features()" is not able to detect more than
+#       one "force_ciphersuite" per client/server and it only picks the 2nd one.
+#       Therefore the 1st one is added explicitly here
```

Comment:
I don't understand this comment: `detect_required_features` runs independently on the client command line and on the server command line. If they have different `force_ciphersuite` options, this should lead to skipping the test case unless both `TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA256` and `TLS-ECDHE-RSA-WITH-AES-128-CBC-SHA256` are enabled. So the explicit requirement on `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` should not be necessary. What am I missing?

Reply:
I just had a look and I don't think you're missing anything. @valeriosetti wdyt?

Reply:
It's been some time since I did this change, so I'm trying to remember the exact reason for it. I looked at the code and I think the problem with these tests:
- `SSL async private: cancel after start then fall back to transparent key`
- `SSL async private: sign, error in resume then fall back to transparent key`
is that they define 2 different ciphersuites for the client (because the client is run twice).

In the scenario you described (which is the most used case) we have 1 ciphersuite for the server and 1 for the client, so `detect_required_features` works well. Here the problem is that we are running the client twice, with different ciphersuites in each case, but `detect_required_features()` only picks the 2nd one (ignoring `ECDHE-ECDSA`). That's why I enforced that in those tests.
I agree that a better approach would have been to extend `detect_required_features()` in order to support multiple ciphersuites, but it was only a matter of 2 tests, so that's why I didn't try to do so.

Reply:
Oh duh, yes, I see, it's because we run the client twice. I agree with not making the detection code more complex just for two test cases. Thanks for the explanation.